File: leetcode/src/test/java/org/luvx/leetcode/java/medium/_129/SolutionTest.java
package org.luvx.leetcode.java.medium._129;

import org.junit.Test;
import org.luvx.leetcode.TreeNode;

public class SolutionTest {
    /**
     * <pre>
     *      1
     *     2  3
     *    4 5 6 7
     * </pre>
     * 124 + 125 + 136 + 137
     */
    @Test
    public void sumNumbersTest() {
        Solution exec = new Solution();
        TreeNode root = new TreeNode(1, new TreeNode(2, new TreeNode(4), new TreeNode(5)), new TreeNode(3, new TreeNode(6), new TreeNode(7)));
        int i = exec.sumNumbers0(root);
        System.out.println(i);
    }
}

File: leetcode/src/test/kotlin/org/luvx/leetcode/java/medium/_94/SolutionTest.kt
package org.luvx.leetcode.java.medium._94;

import org.luvx.leetcode.tree.TreeNode
import org.luvx.coding.common.enhancer.out
import kotlin.test.BeforeTest
import kotlin.test.Test

class SolutionTest {
    private var root: TreeNode? = null

    @BeforeTest
    fun before() {
        root = TreeNode.of(
            5,
            TreeNode.of(
                4,
                TreeNode.of(
                    11,
                    TreeNode.of(7),
                    TreeNode.of(2)
                )
            ),
            TreeNode.of(
                8,
                TreeNode.of(13),
                TreeNode.of(
                    4, null,
                    TreeNode.of(1)
                )
            )
        )
    }

    /**
     * <pre>
     *          5
     *         / \
     *        4   8
     *       /   / \
     *      11  13  4
     *     /  \      \
     *    7    2      1
     * </pre>
     */
    @Test
    fun hasPathSumTest() {
        val exec = Solution()
        exec.inorderTraversal(root).out()
        exec.inorderTraversal1(root).out()
    }
}